subsys/bluetooth/controller/hal/nrf5/rand.c

```diff
@@ -12,36 +12,60 @@
 #include "common/log.h"
 #include "hal/debug.h"
 
-#define RAND_RESERVED (4)
+enum rand_status {
+	RAND_STATUS_CONTINUE,
+	RAND_STATUS_DONE,
+	RAND_STATUS_FULL,
+};
 
 struct rand {
 	u8_t count;
+	u8_t threshold;
 	u8_t first;
 	u8_t last;
 	u8_t rand[1];
 };
 
-static struct rand *rng;
+static struct rand *rng_isr;
+static struct rand *rng_thr;
 
-void rand_init(u8_t *context, u8_t context_len)
+static void _init(struct rand **rng, u8_t *context, u8_t len, u8_t threshold)
```

Comment:
_ underscore is reserved for the kernel, let's avoid it from now on even if it's static